### DISCUSSION TITLE
Managing etcd users and permissions using IaC tools?

### DISCUSSION THREAD

**Discussion by rc-donalhunt** [etcd-io/etcd#16172] [Q&A]
Is anyone using an IaC tool (ansible, terraform) for managing etcd users and/or permissions? 

I've done some searching and doesn't look like there is any modules or roles that support this right now. My current thinking is to be able to manage the users and permissions in a configuration management config file and then provision the user / permissions from those configs in line with how we manage other RMAC-based systems.

How are others approaching this problem?

**Comment by jmhbnz** [comment]
Just a short note that I agree with your findings, I haven't come across any IaC tooling for managing etcd users.  There are plenty of roles for installing etcd initially or fetching values, example: https://docs.ansible.com/ansible/latest/collections/community/general/etcd3_module.html 

I imagine many folks just have some automation around `etcdctl` or something similar: https://etcd.io/docs/v3.3/op-guide/authentication/#working-with-users
